lib/smb2/dispatcher/socket.rb

```diff
@@ -7,7 +9,7 @@ class Smb2::Dispatcher::Socket < Smb2::Dispatcher::Base
   # @param host [String] passed to TCPSocket.new
   # @param port [Fixnum] passed to TCPSocket.new
   def self.connect(host, port)
-    new(TCPSocket.new(host, port))
+    new(::TCPSocket.new(host, port))
```

Comment:
Will the socket class need to be injected for usage in metasploit-framework to support tunnelled sockets?